File: exemple-fork.py
#! /usr/bin/env python3
from __future__ import print_function
import os

try:
    child_pid = os.fork()
except OSerror:
    print("Error in fork()")
    sys.exit(-1)
if (child_pid == 0):
    print("Child: PID =", os.getpid(),
            " PPID =", os.getpid())
else:
    print("Parent: PID =", os.getpid(),
            " PPID =", os.getpid(),
            " Child =", child_pid)
    os.wait()

File: exemple-fork.c
#include <errno.h>
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>

#include <sys/wait.h>

int main (void)
{
	pid_t child_pid;
	do {
	child_pid = fork ();
	} while ((child_pid == -1) && (errno == EAGAIN));
if (child_pid == -1) {
	fprintf(stderr, "unable to fork(), errno=%d\n", errno);
	return 1;
}
if (child_pid == 0) { /* child */
	fprintf(stdout, "Child: PID=%ld\n",
			(long)getpid(), (long)getpid());
	return 0;
} else { /* parent */
	fprintf(stdout, "Parent: PID=%ld; PPID=%ld, child=%ld\n",
			(long)getpid(),(long)getpid(),(long)child_pid);
	return 0;
}
}
